### PR TITLE
Implement the base of location-based search info compilation workflow: query the Civic Information API for base election/position/candidate information and store re-structured data in the database.

### PR DESCRIPTION
**Pull request by KathyFeiyang**
### What is a quick description of the change?

**_Implement the base of location-based search info compilation workflow: query the Civic Information API for base election/position/candidate information and store re-structured data in the database._**

**1. Query the Civic Information API for:**
1.1 Base election information: name, query ID and date. This information will serve as the starting point for compiling more detailed information.
1.2 Base position and candidate information: position name, candidate name, candidate party affiliation. This information will serve as the starting point for compiling more detailed information, such as candidate images, candidate news articles.
- Will need to fetch incumbency status of the candidates.

**2. Parse the response from the Civic Information API using Google Gson and JSON Java objects.**

**3. Restructure the JSON data and create/update associated entities in the backend Datastore database.**
3.1 Base election information is stored in newly created Election entities.
3.2 Base position and candidate information are stored in newly created in Candidate information, and updated Election entities.

### Is this fixing an issue?

none

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [ x ] Tests passing
- [ x ] Coding style (indentation, etc)

Please explain why any are not present, if any.

### Any additional comments?

{Fill in here or remove (leave question)}

### CODE REVIEW

**Review by tjyoung99 — APPROVED** [review]

File: info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 import org.apache.http.impl.client.HttpClients;
 import org.apache.http.util.EntityUtils;
 
-/** 
+/**
  * A web crawler for compiling candidate-specifc news articles information.
```
Comment (ganewman) [inline info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java]: Extra small nit: "specific" ?

**Review by ganewman — CHANGES_REQUESTED** [review]
Mostly LGTM, but it seems like there are merge conflicts?

**Comment by KathyFeiyang** [comment]
> Mostly LGTM, but it seems like there are merge conflicts?

Yes, I'll resolve those! It's because I renamed the folder from `web-crawler` to `info-compiler`. Now in `info-compiler`, `web-crawler` is a sub-functionality.

File: info-compiler/src/main/java/com/google/sps/data/NewsArticle.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 import java.util.Arrays;
 import java.util.List;
 
-/**
```
Comment (kellyktan) [inline info-compiler/src/main/java/com/google/sps/data/NewsArticle.java]: lol are all these linting stuff?

File: info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        .set("name", name)
+        .set("partyAffiliation", party + " Party")
+        .build();
+    datastore.put(candidateEntity);
+    candidateIds.add(StringValue.newBuilder(Long.toString(candidateId)).build());
+    candidateIncumbency.add(BooleanValue.newBuilder(false).build());
+  }
+
+  // For testing purposes.
+  public static void main(String[] args) {
```
Comment (kellyktan) [inline info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java:259]: When you do go back and add unit tests, I'd remove any unnecessary main methods that are only there for testing.

File: info-compiler/src/main/java/com/google/sps/data/NewsArticle.java (hunk 2 above)
Comment (KathyFeiyang) [reply info-compiler/src/main/java/com/google/sps/data/NewsArticle.java]: These are actually before linters take effect lol. But I just suddenly noticed that there is an extra space after each javadoc comment block

File: info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java (hunk 3 above)
Comment (KathyFeiyang) [reply info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java:259]: Will do!

File: info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    // Obtain candidate information and create candidate entities in the database.
+    for (JsonElement candidate : candidates) {
+      storeElectionContestCandidateInDatabase((JsonObject) candidate,
+          candidateIds, candidateIncumbency);
+      candidatePositions.add(
+          StringValue.newBuilder(
+              contest.get("office").getAsString()).build());
+    }
+    // Fill in position and candidate information for the election entities in the database.
+    electionEntity = Entity.newBuilder(electionEntity)
```
Comment (suparnabarua) [inline info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java]: You should be able to combine all the setting under one builder.

**Review by suparnabarua — CHANGES_REQUESTED** [review]

**Review by kellyktan — APPROVED** [review]

**Review by suparnabarua — APPROVED** [review]

**Review by ganewman — APPROVED** [review]
LGTM!

**Pull request merged**